cmd_line_tools/commands.py

```diff
@@ -1,11 +1,195 @@
+    and then outputs them
+    """
+    REQUEST_URL = 'http://pokeapi.co/api/v2/berry/'
+
+    def get_pokemon(self):
+        return self.request_data()
+
+
+class PokemonBerryPrinter(PagedJSONDataMixin,
+                          StdoutOutputMixin,
+                          BaseAllPokemonBerriesCommand):
+    """ Prints to stdout """
+
+    def main(self):
+        results = self.get_pokemon()
+        for berry in results:
+            self.write(berry['name'])
+
+
+class PokemonBerryCSVWriter(PagedJSONDataMixin,
+                            CSVOutputMixin,
+                            BaseAllPokemonBerriesCommand):
+    FILE_PATH = 'pokemonberries.csv'
+    """ Exports to CSV file """
+
+    def main(self):
+        results = self.get_pokemon()
+        for berry in results:
+            self.write(berry)
+
+
+class BasePokemonCommand(object):
+    """
+    Retrieves info on a specific pokemon from PokeAPI
+    Works for both pokemon name and id in the api
+    ID/Name may be passed or will be requested
+
+    Output as following:
+    Name: Pokemon Name
+    Moves:
+    move 1
+    move 2
+    ...
+    last move
+    """
+
+    URL = 'http://pokeapi.co/api/v2/pokemon/{}'
+
+    OUTPUT_FORMAT = "\nName: {}\nMoves:\n{}"
+
+    def get_pokemon(self, name=None):
+        if name is None:
+            name = self.request_input_data('pokemon_name')
+        self._get_data(name)
+        return self._format()
+
+    def _get_data(self, id):
+        self.REQUEST_URL = self.URL.format(id)
+        self.data = self.request_data()
+
+    def _format(self):
+        if self.data:
+            moves = ""
+            for move in self.data['moves']:
+                moves += move['move']['name'] + "\n"
+
+            return self.OUTPUT_FORMAT.format(self.data['name'], moves)
+        else:
+            return ""
+
+
+class InputBasedPokemon(SimpleCommandLineParserMixin,
+                        InputRequestMixin,
+                        JSONDataRequestMixin,
+                        StdoutOutputMixin,
+                        BasePokemonCommand):
+    """ Retrieves pokemon based on user input """
+
+    def main(self):
+        result = self.get_pokemon()
+        self.write("Result: {}".format(result))
+
+
+class CommandLinePokemon(SimpleCommandLineParserMixin,
+                         ArgumentsRequestMixin,
+                         JSONDataRequestMixin,
+                         StdoutOutputMixin,
+                         BasePokemonCommand):
+    """ Retrieves pokemon based on command line argument """
+
+    def main(self):
+        self.parse_arguments()
+        result = self.get_pokemon()
+        self.write("Result: {}".format(result))
+
+
+class RandomChooserPokemon(JSONDataRequestMixin,
+                           StdoutOutputMixin,
+                           BasePokemonCommand):
+    """
+    Retrieves information of random pokemon
+    N_POKEMON = Number of pokemon available from PokeAPI
+    """
+    N_POKEMON = 811
```

Comment:
Seems would have been better to get this value from the API rather than hard coding it.